src/hooks/usePagination.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-explicit-any*/
+
+import { GitHubPaginatedResponse } from "@/lib/github/apiActions/baseApiActions";
+import { useState } from "react";
+
+// A custom hook to manage pagination state and navigation for paginated GitHub API calls
+export function usePagination<T>() {
+  const [nextPage, setNextPage] = useState<number | undefined>(undefined);
+  const [previousPage, setPreviousPage] = useState<number | undefined>(
+    undefined
+  );
```

Comment:
```suggestion
  const [previousPage, setPreviousPage] = useState<number>();
```

This does the same thing with less code. Providing no initial value tells [TypeScript/React/the TS compiler/ idk exactly] that the value can be `undefined` without you including it in the type. Something like `useState<number | undefined>` is helpful when you want to provide an initial value, because in that case [the powers that be] would _not_ know that the value is allowed to be undefined.